nautobot/extras/utils.py

```diff
@@ -80,6 +80,52 @@ def get_choices(self):
         return [(f"{ct.app_label}.{ct.model}", ct.pk) for ct in ContentType.objects.filter(self.get_query())]
 
 
+@deconstructible
+class PrimaryModelRelatedContentType:
```

Comment:
I wonder if it'd be possible to make this more generic and reusable, so that it could use any desired class as the base class. Taking `FeatureQuery` as an example, it would be pretty handy to be able to use this via something like `ModelSubclassesQuery(PrimaryModel)`, `ModelSubclassesQuery(OrganizationalModel)`, `ModelSubclassesQuery(RelationshipModel)`, etc.

Reply:
Sure

Reply:
I can't currently archive this because of a circular import error raise  when import PrimaryModel in nautobot/extras/models/tags.py